### ISSUE TITLE
Node Debugger not working after updating to June 2020 (version 1.47)

### ISSUE THREAD

**Issue by harshal123** [microsoft/vscode-js-debug#631]
**Describe the bug**
Node Debugger not working after updating to June 2020 (version 1.47)

**To Reproduce**
After update when I tried to run a server in debug mode it shows me the "options [retryMiliSeconds] is not supported". Server gets started but debugger points gets disappeared.   

**Log File**

**VS Code Version:** Replace me!

**Additional context**

**Comment by connor4312** [comment]
Can you please collect a log file using the instructions provided in the issue template?

**Comment by harshal123** [comment]
Hi,

I can see this

Process exited with code 1
C:\Program Files (x86)\Nodist\bin\node.exe c:\work\integration-api\index.js
Debugger listening on ws://127.0.0.1:65181/46baf111-1335-454a-ab78-72ed2dc308aa
For help see https://nodejs.org/en/docs/inspector
the options [retryMiliSeconds] is not supported
(node:3428) [DEP0016] DeprecationWarning: 'GLOBAL' is deprecated, use
'global'
the options [retryMiliSeconds] is not supported

**Comment by jasonwilliams** [comment]
Same for me, its just stopped working since this release, i get "unbound breakpoint" whereever i set it :(

**Comment by Doubl3** [comment]
**Doesn't work anymore as well**
- VSCode version
```
Version: 1.47.2
Commit: 17299e413d5590b14ab0340ea477cdd86ff13daf
Date: 2020-07-15T18:18:50.054Z (1 day ago)
```

- Command launched in a terminal:
```
[nodemon] 1.19.4
[nodemon] to restart at any time, enter `rs`
[nodemon] watching dir(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node --inspect-brk=0 --debug=3xxx src/service.js`
Debugger listening on ws://127.0.0.1:3xxx/51879355-9a6a-435e-90a8-f9f84f29c6a7
```

- Cfg used to debug NodeJS process on VSCode side (this cfg worked with previous VSCode version):
```
{
      "type": "node",
      "request": "attach",
      "name": "Process name",
      "port": 3xxx
    },
```

**Comment by connor4312** [comment]
Can you collect a trace log using these instructions?

> If you're able to, add `"trace": true` to your `launch.json` and reproduce the issue. The location of the log file on your disk will be written to the Debug Console. Share that with us.
>
> ⚠️ This log file will not contain source code, but will contain file paths. You can drop it into https://microsoft.github.io/vscode-pwa-analyzer/index.html to see what it contains. If you'd rather not share the log publicly, you can email it to connor@xbox.com

**Comment by jamesgibson14** [comment]
Just a couple of things I have noticed since the update (I develop with Meteor 1.10): 

1. Auto Attach no longer works (it never connects), I have to use the command "Attache to Node Process"
2. Every new terminal (bash) I open runs a debugger by default, I use a lot of terminals so this is a bug for me.
3. I ran across this error with one of my Meteor apps which cause it to fail: Error: Meteor cannot currently handle quoted NODE_OPTIONS

**Comment by connor4312** [comment]
@jamesgibson14 see https://github.com/microsoft/vscode/issues/102057#issuecomment-657728467. You can reset this behavior by:

1. Making sure you're in 1.47.2
2. Toggling auto attach to "disabled" and then back "on" (if you'd like to use it)

**Comment by connor4312** [comment]
Also @Doubl3 I notice you have some args `--inspect-brk=0 --debug=3xxx`. Try removing these and seeing if that works for you; we no longer need these to be able to debug processes.

If you do _need_ to attach to that exact port, then you can use `attachSimplePort` https://github.com/microsoft/vscode-js-debug/issues/586#issuecomment-658958272

**Comment by jamesgibson14** [comment]
@connor4312 thanks, I changed those configurations and so far the auto attach seems to be working again.

**Comment by illusional** [comment]
Piggybacking on this issue, the update broke my node (TypeScript) debugging too:

For me, it couldn't find a module:

```
Cannot find module "mongodb-extjson"
```

**launch.json**

```json
{
    "type": "node",
    "request": "launch",
    "name": "Launch Program",
    "program": "${workspaceFolder}/dist/server.js",
    "preLaunchTask": "tsc: build - tsconfig.json",
    "outFiles": [
        "${workspaceFolder}/dist/**/*.js"
    ],
    "env": {
        "NODE_ENV": "development"
    }
},
```

Launching in `Version: 1.46.1` works fine, but the June update broke it specifically. I'm still able to run through the command line, but debugging is broken.

**Comment by connor4312** [comment]
Can you collect a trace log using these instructions? https://github.com/microsoft/vscode-js-debug/issues/631#issuecomment-660149131

**Comment by illusional** [comment]
Hmm, I downgraded and reupgraded and it's now working. So I'll retract my original statement. Not really sure why but glad it's working now.

**Comment by connor4312** [comment]
Okay, thanks for the info. I suspect the issue was not related to the debugger, perhaps some environment or caching problem somewhere.

**Comment by sageguy09** [comment]
Worth noting we are seeing an issue as well. We  use NPM and have a a debug run-script that looks like this :
```
"debug": "cross-env NODE_ENV=development node --nolazy --inspect-brk=5858 index.js"
```
and a launch.json config like :

```
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Launch via NPM",
            "cwd" : "${workspaceFolder}",
            "runtimeExecutable": "npm",
            "runtimeArgs": [
                "run-script", "debug"
            ],
            "port": 5858
        }
```

The debugger doesn't fully attach, our console.log() statements in `index.js` aren't hit, and breakpoints light up. Looks like the call stack is stuck on cross-env and never hits index.js. 

(We are using the NPM configuration listed on the Debug documentation on VS Code docs.)

Let me know if we can provide any further assistance debugging.

**Comment by connor4312** [comment]
Can you collect a trace log using these steps? https://github.com/microsoft/vscode-js-debug/issues/631#issuecomment-660149131

**Comment by sageguy09** [comment]
Issues (while using the configuration listed above) are still ongoing. I have emailed the trace file requested.

**Comment by connor4312** [comment]
Thanks for the log!

I think I already fixed your issue as part of https://github.com/microsoft/vscode-js-debug/commit/540fbcace9eda0c43fa0b73797d3cba95a2f443b. I think both of these solutions will work for you:

1. Use the [nightly build](https://github.com/microsoft/vscode-js-debug#nightly-extension)
1. Remove `--inspect-brk` from your npm script (we can attach without it; I need to update our docs to reflect that)

**Comment by hootyjeremy** [comment]
> Thanks for the log!
> 
> I think I already fixed your issue as part of [540fbca](https://github.com/microsoft/vscode-js-debug/commit/540fbcace9eda0c43fa0b73797d3cba95a2f443b). I think both of these solutions will work for you:
> 
> 1. Use the [nightly build](https://github.com/microsoft/vscode-js-debug#nightly-extension)
> 2. Remove `--inspect-brk` from your npm script (we can attach without it; I've updated the docs to reflect that in [microsoft/vscode-docs@26b07f5](https://github.com/microsoft/vscode-docs/commit/26b07f527e9862827492aa8fb604dec13a5ef824))

Hello.

I think I am very close here. I have gotten nodejs to properly begin debugging my app after removing the --inspect--brk, but the debugger doesn't seem to look at the port specified in my launch.json. It will just choose a random port. This means that I can't do any remote debugging with Chrome's DevTools unless I specify a listening port every time. I like being able to use that tool as well as the vscode tools. Here is the launch.json...

`{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "Launch Program",
            "skipFiles": [
                "<node_internals>/**"
            ],
            "program": "${workspaceFolder}\\pubg.js",
            "runtimeExecutable": "npm",
            "runtimeArgs": ["run-script", "debug"],
            "port": 9229
        }
    ]
}`

And here is the output as I begin debugging with the launch...

> "E:\Program Files\nodejs\npm.cmd" run-script debug "c:\Users\hooty\Desktop\Hooty Pubg Api js (0.011)\pubg.js"
Debugger listening on ws://127.0.0.1:58080/a1829a9e-f611-434e-a000-61750dc012b6
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.

C:\Users\hooty\AppData\Roaming\npm
e:\Program Files\nodejs\node_modules\npm\lib\utils\output.js:6
Waiting for the debugger to disconnect...
Debugger listening on ws://127.0.0.1:58087/d57e6182-f9a1-4a9f-a4e2-4b8c19d7b9e3
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.

> hootypubg@1.0.0 debug c:\Users\hooty\Desktop\Hooty Pubg Api js (0.011)
> node pubg.js "c:\Users\hooty\Desktop\Hooty Pubg Api js (0.011)\pubg.js"

Anyway, just wanted to share. It doesn't seem to be looking at the port in the launch.json. Once it does that, I'll be solid.

Thanks.

**Comment by connor4312** [comment]
Closing this as we've not heard back from the author with details.
